services/analyzer/mongo-collections-analyzer.js

```diff
@@ -4,6 +4,18 @@ const logger = require('../logger');
 const { DatabaseAnalyzerError } = require('../../utils/errors');
 const { detectReferences, applyReferences } = require('./mongo-references-analyzer');
 const { detectHasMany, applyHasMany } = require('./mongo-hasmany-analyzer');
+const {
+  getMongooseTypeFromValue,
+  isOfMongooseType,
+} = require('../../utils/mongo-primitive-type');
+const {
+  getMongooseArraySchema,
+  getMongooseEmbeddedSchema,
+  getMongooseSchema,
```

Comment:
[IGNORABLE] we are in mongo prefixed files so no need to let mongo in function.

Reply:
I would rather keep this naming, because what we want is mongoose specific data, not even mongo data. IMO the mongoose word has its own meaning